pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -62,18 +62,19 @@ var (
 	accountsLock = &sync.Mutex{}
 )
 
-func newBlobDiskController(common *controllerCommon) (*BlobDiskController, error) {
-	c := BlobDiskController{common: common}
+func (c *BlobDiskController) initStorageAccounts() {
+	accountsLock.Lock()
+	defer accountsLock.Unlock()
 
-	// get accounts
-	accounts, err := c.getAllStorageAccounts()
-	if err != nil {
-		klog.Errorf("azureDisk - getAllStorageAccounts error: %v", err)
-		c.accounts = make(map[string]*storageAccountState)
-		return &c, nil
+	if c.accounts == nil {
+		// get accounts
+		accounts, err := c.getAllStorageAccounts()
+		if err != nil {
+			klog.Errorf("azureDisk - getAllStorageAccounts error: %v", err)
+			c.accounts = make(map[string]*storageAccountState)
```

Comment:
Keep c.accouts as nil here? or else, it won't try again on errors.

Reply:
@feiskyer yes, won't try again on error, it would make new map and create new storage account(don't use existing accounts)